tests/e2e/config_secret_utils.go

```diff
@@ -0,0 +1,331 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	"golang.org/x/crypto/ssh"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fssh "k8s.io/kubernetes/test/e2e/framework/ssh"
+)
+
+var insecureFlag bool
+var clusterId string
+var clusterDistribution string
+var dataCenters string
+var vCenterPort string
+var user string
+var password string
+var vCenterIp string
+
+func govcLoginCmd(vcAddress string) string {
+	loginCmd := "export GOVC_INSECURE=1;"
+	loginCmd += fmt.Sprintf("export GOVC_URL='https://administrator@vsphere.local:Admin!23@%s';", vcAddress)
+	framework.Logf(loginCmd)
+	return loginCmd
+}
+
+func createTestUser(loginCmd string, sshClientConfig *ssh.ClientConfig, masterIp string,
+	testUser string, testUserPassword string) error {
+	// create 2 test users
+	createUser := loginCmd + "govc sso.user.create -p " + testUserPassword + " " + testUser
+	framework.Logf(createUser)
+	result, err := sshExec(sshClientConfig, masterIp, createUser)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			createUser, masterIp, err)
+	}
+	return nil
+}
+
+func deleteUsersRolesAndPermissions(loginCmd string, sshClientConfig *ssh.ClientConfig,
```

Comment:
you can keep this method, but also create smaller routines for each of the parts i.e., have different methods for deleting roles, permissions, users and have a wrapper to call all of them

Reply:
done